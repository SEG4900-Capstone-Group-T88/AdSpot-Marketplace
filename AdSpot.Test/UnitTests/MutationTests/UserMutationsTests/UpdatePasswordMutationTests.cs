namespace AdSpot.Test.UnitTests.UserMutationsTests;

[Collection("adspot-inmemory-db")]
public class UpdatePasswordMutationTests
{
    private const string UpdatePasswordMutation = """
            mutation UpdatePassword($input: UpdatePasswordInput!) {
                updatePassword(input: $input) {
                    user {
                        userId
                        email
                        firstName
                        lastName
                    }

                    errors {
                        ... on Error {
                            __typename
                            message
                        }
                    }
                }
            }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task UpdatePasswordSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetDocument(UpdatePasswordMutation)
                .SetVariableValues(
                    new Dictionary<string, object?>
                    {
                        {
                            "input",
                            new Dictionary<string, object?>
                            {
                                { "userId", TestDatabase.TestUser.UserId },
                                { "password", "UpdatedTestPassword" }
                            }
                        }
                    }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task UpdatePasswordInvalidUserIdError()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetDocument(UpdatePasswordMutation)
                .SetVariableValues(
                    new Dictionary<string, object?>
                    {
                        {
                            "input",
                            new Dictionary<string, object?> { { "userId", 2 }, { "password", "UpdatedTestPassword" } }
                        }
                    }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }
}
