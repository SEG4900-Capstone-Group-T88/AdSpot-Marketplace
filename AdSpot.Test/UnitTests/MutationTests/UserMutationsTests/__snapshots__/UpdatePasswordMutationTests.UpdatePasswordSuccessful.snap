{
  "data": {
    "updatePassword": {
      "user": {
        "userId": 1,
        "email": "testuser@adspot.com",
        "firstName": "Test",
        "lastName": "User"
      },
      "errors": null
    }
  }
}